src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        table = resize();
+        int hash = (key == null) ? 0 : key.hashCode();
+        int bucket = Math.abs(hash) % capacity;
+        Node<K, V> node = table[bucket];
+        Node<K, V> newNode;
+        if (node == null) {
+            table[bucket] = new Node<>(hash,key,value, null);
+            size++;
+            return;
+        }
+        while (node != null) {
+            if (hash == node.hash && node.key == null ? key == null : node.key.equals(key)) {
+                node.value = value;
+                return;
+            } else if (node.next == null) {
+                newNode = new Node<>(hash,key,value, null);
+                node.next = newNode;
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        table = resize();
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode());
+        int bucket = hash % capacity;
+        Node<K, V> node = table[bucket];
+        if (node == null) {
+            return null;
+        }
+        while (node != null) {
+            if (hash == node.hash && node.key == null ? key == null : node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V>[] resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : oldTable.length;
+        int oldThreshold = threshold;
+        if (oldThreshold == 0) {
+            threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+            capacity = DEFAULT_INITIAL_CAPACITY;
+            return new Node[capacity];
+        }
+        if (threshold > (size + 1)) {
+            return oldTable;
+        }
```

Comment:
```suggestion
        if (threshold > (size + 1)) {
            return oldTable;
        }
```
I suggest move check threshold to put